spring-cloud-function-stream/src/main/java/org/springframework/cloud/function/stream/SupplierInvokingMessageProducer.java

```diff
@@ -33,24 +31,30 @@
  */
 public class SupplierInvokingMessageProducer<T> extends MessageProducerSupport {
 
-	private final Supplier<Flux<T>> supplier;
+	private final FunctionCatalog functionCatalog;
 
-	public SupplierInvokingMessageProducer(Supplier<?> supplier, long interval) {
```

Comment:
we're losing the interval support here, but I suppose that should be added via yet another wrapper level above this